Modix.Services/Starboard/StarboardService.cs

```diff
@@ -0,0 +1,201 @@
+
+        /// <summary>
+        /// Gets the appropriate star-emote for the given <paramref name="reactionCount"/>.
+        /// </summary>
+        /// <param name="reactionCount"></param>
+        /// <returns>A star-emote in string format</returns>
+        string GetStarEmote(int reactionCount);
+
+        /// <summary>
+        /// Modifies a starboard entry.
+        /// </summary>
+        /// <param name="guild">Which guild's starboard to operate on</param>
+        /// <param name="message">The message to modify</param>
+        /// <param name="content">The content to modify with</param>
+        /// <returns>A <see cref="Task"/> that will complete when the operation has completed.</returns>
+        Task ModifyEntry(IGuild guild, IUserMessage message, string content);
+
+        /// <summary>
+        /// Adds an <see cref="IUserMessage"/> to the starboard
+        /// </summary>
+        /// <param name="guild">The guild on which starboard to post</param>
+        /// <param name="message">The message to add to the starboard</param>
+        /// <param name="content">Meta-data that contains relevant information about the starred message</param>
+        /// <param name="embed">The embed to include in the starboard-message</param>
+        /// /// <returns>A <see cref="Task"/> that will complete when the operation has completed.</returns>
+        Task AddToStarboard(IGuild guild, IUserMessage message, string content, Embed embed);
+    }
+
+    /// <inheritdoc />
+    public class StarboardService : IStarboardService
+    {
+        private readonly IDesignatedChannelService _designatedChannelService;
+        private readonly IMessageRepository _messageRepository;
+        private static readonly IReadOnlyDictionary<int, string> _emojis = new Dictionary<int, string>
+        {
+            { 20, "✨"},
+            { 10, "💫" },
+            { 5, "🌟" },
+            { 0, "⭐" }
+        }.OrderByDescending(k => k.Key).ToDictionary(k => k.Key, k => k.Value);
+
+        public StarboardService(
+            IDesignatedChannelService designatedChannelService,
+            IMessageRepository messageRepository)
+        {
+            _designatedChannelService = designatedChannelService;
+            _messageRepository = messageRepository;
+        }
+
+        /// <inheritdoc />
+        public async Task<bool> ExistsOnStarboard(IMessage message)
+        {
+            var messageEntity = await _messageRepository.GetMessage(message.Id);
+            return messageEntity?.StarboardEntryId != null;
+        }
+
+        private async Task<IUserMessage> GetStarboardEntry(IGuild guild, IMessage message)
+        {
+            var channel = await GetStarboardChannel(guild);
+            var entity = await _messageRepository.GetMessage(message.Id);
+            return await channel.GetMessageAsync(entity.StarboardEntryId.Value) as IUserMessage;
+        }
+
+        private async Task<ITextChannel> GetStarboardChannel(IGuild guild)
```

Comment:
Documentation?

Reply:
I semi-agree with you on documenting any public function if it isn't self-explanatory, but private functions, not so much.

Reply:
I agree, the code should speak for itself, but documentation is helpful to be truly transparent.